components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/TenantThemeApiServiceImpl.java

```diff
@@ -16,43 +16,98 @@
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
 import org.wso2.carbon.apimgt.rest.api.admin.v1.TenantThemeApiService;
 import org.wso2.carbon.apimgt.rest.api.admin.v1.utils.RestApiAdminUtils;
+import org.wso2.carbon.apimgt.rest.api.util.RestApiConstants;
 import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
 
+import java.io.File;
+import java.io.IOException;
 import java.io.InputStream;
 
+import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 public class TenantThemeApiServiceImpl implements TenantThemeApiService {
 
     private static final Log log = LogFactory.getLog(TenantThemeApiServiceImpl.class);
+    private static final String TENANT_THEMES_EXPORT_DIR_PREFIX = "exported-tenant-themes";
 
     /**
-     * Import an Tenant Theme for a particular tenant by uploading an archive file.
+     * Import a Tenant Theme for a particular tenant by uploading an archive file.
      *
      * @param fileInputStream content relevant to the tenant theme
      * @param fileDetail      file details as Attachment
+     * @param messageContext
      * @return Theme import response
+     * @throws APIManagementException if an error occurs when importing a tenant theme
      */
     @Override
-    public Response tenantThemeImportPost(InputStream fileInputStream, Attachment fileDetail,
-                                          MessageContext messageContext) {
+    public Response importTenantTheme(InputStream fileInputStream, Attachment fileDetail, MessageContext messageContext)
+            throws APIManagementException {
 
         String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+        if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            String errorMessage = "Super Tenant " + MultitenantConstants.SUPER_TENANT_DOMAIN_NAME +
+                    " is not allowed to import a tenant theme";
+            throw new APIManagementException(errorMessage,
+                    ExceptionCodes.from(ExceptionCodes.TENANT_THEME_IMPORT_NOT_ALLOWED,
+                            MultitenantConstants.SUPER_TENANT_DOMAIN_NAME));
+        }
+
         try {
-            RestApiAdminUtils.deployTenantTheme(fileInputStream, tenantDomain);
+            RestApiAdminUtils.importTenantTheme(fileInputStream, tenantDomain);
             return Response.status(Response.Status.OK).entity("Theme imported successfully").build();
-        } catch (APIManagementException e) {
-            String errorMessage = "Error while importing tenant theme for tenant " + tenantDomain;
-            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        } catch (IOException e) {
+            throw new APIManagementException(e.getMessage(),
+                    ExceptionCodes.from(ExceptionCodes.TENANT_THEME_IMPORT_FAILED, tenantDomain, e.getMessage()));
```

Comment:
Shall we propergate the "e" object as well? Otherwise the stacktrace will be swallowed.

Reply:
Please fix this in other places as well.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/8927/commits/5c1474fe6566ea5d341158fd530da08ab4e6af0b